packages/workspace-utils/src/index.ts

```diff
@@ -115,6 +116,54 @@ export async function getWorkspaceGradedFiles(
   return files;
 }
 
+/**
+ * Updates the disk usage of a workspace. This is computed as the sum of the
+ * sizes of all versions of the workspace. The result is stored in the `disk_usage`
+ * column of the `workspaces` table. The total size is returned.
+ *
+ * @param workspace_id The ID of the workspace to update.
+ * @param workspacesRoot The root directory of all workspace data.
+ */
+export async function updateWorkspaceDiskUsage(
+  workspace_id: string,
+  workspacesRoot: string,
+): Promise<number> {
+  const result = await queryOneRowAsync(sql.select_workspace, { workspace_id });
+  const workspace = result.rows[0];
+
+  // We'll compute the size for all versions of the workspace so that we don't need
+  // to separately store the size for each version.
```

Comment:
In reality I think it'd actually be ideal to store this on a per-version basis so we don't always have to waste time computing the size of old versions. 🤷

Reply:
True, but in practice this hopefully won't matter much.